### ISSUE TITLE
Smelting time shown on items

### ISSUE THREAD

**Issue by HodajuciDjed** [DinoDevs/GladiatusCrazyAddon#350]
Hello, fist of I love all the work that's done to improve the quality of the game with the addon significantly! Good job!

Is there any chance you can put **smelting time** for **items** on their tooltip somewere like the materials or just outright below the level? _(I have added pictures to better illustrate what I meant)_

I am a level 120 and if you actively play you know the strugle of 5-10 pages of smelting items in packages.
It would be a huge time saver because I need to move items to my already "full" bags to see if items are done smelting by the morning or by the time I get of work, a small thing but I think a great addition as this way I can choose them in the packages and I don't need to repack them to save space.

![paketi](https://user-images.githubusercontent.com/69336801/166121852-10d5e26a-3bd3-47e8-84d2-dc8501a4e50d.png)
![tooltip](https://user-images.githubusercontent.com/69336801/166121869-925f723f-9fd5-4854-a4d2-67e8366b9b66.png)

Best regards!

**Comment by FrutyX** [comment]
I can't promise that it will be added (in some way), but I like the idea! I also keep having too much stuff to smelt, this would help.

**Comment by GramThanos** [comment]
We have to find the formula first.
I expect it to be something like:

```
seconds = ((item-level * item-type) * item-quality) / server-speed
```

**Comment by HodajuciDjed** [comment]
Sweet, glad you like it and find it usefull. I understand isn't not a priority it's a small thing just something to keep in mind for future updates :D

**Comment by GreatApo** [comment]
From a quick data analysis, the smelting time seems to be related to the item value:
total time in seconds = value * 0.93745

Can anybody else verify this? And an example on a speed server will be useful...

![image](https://user-images.githubusercontent.com/9620510/175794485-67561e59-d24a-4f30-9ce9-07b34b9c5384.png)

**Comment by FrutyX** [comment]
There is so much data needed for this...

Wouldn't it be easier to do something like this and just simulate it with it (via smelter free slot)?

![hammer](https://user-images.githubusercontent.com/23278552/175812648-c6a8cb4f-0104-4141-8eb6-62879a3c7ae7.PNG)

Of course it's a bit away from the original idea, but, you know.

**Comment by GreatApo** [comment]
Actually if what I found works for all levels, then it is quite easy to add the time in the tooltips

**Comment by SleepingShadow-bit** [comment]
i will check out after the smelting event. 
i am on a x4 server

**Comment by FilipSla** [comment]
The coefficients are similar for me and according to server speed you can just divide/multiply them 😃
![image](https://user-images.githubusercontent.com/44546766/176299259-c65df996-475e-48f8-aebc-4f03d018715b.png)

**Comment by GreatApo** [comment]
> The coefficients are similar for me and according to server speed you can just divide/multiply them 😃

I think that's a green light

**Comment by GreatApo** [comment]
Forging time is double in underworld? We will see if the times still work once I get out...

![image](https://user-images.githubusercontent.com/9620510/176311715-b3fc224a-49f5-4beb-b8dc-fbc9bb8eaff9.png)

**Comment by FrutyX** [comment]
Double? No, that has to be a bug.

**Comment by GreatApo** [comment]
> Double? No, that has to be a bug.

It seems it is double in underword:
![image](https://user-images.githubusercontent.com/9620510/176313357-0acc2a1f-9fa3-4c71-ace9-b7e90c2c0caa.png)
![image](https://user-images.githubusercontent.com/9620510/176313518-2e3e4661-d38b-4be4-8fe0-ffaff4d0216e.png)
(last item is while in underworld)

**Comment by FrutyX** [comment]
Ehm, I am not in the underworld.


![hmmm](https://user-images.githubusercontent.com/23278552/176315692-7a92f469-9b27-4ba5-98f3-4bdfe8e0538a.PNG)

**Comment by GreatApo** [comment]
> Ehm, I am not in the underworld. Speed x5
> 

There was an event with faster forging times, dig I got all my measurements by half? hahaha

**Comment by FrutyX** [comment]
Also I noticed that if you move them item (drag), the time will disappear.

**Comment by GreatApo** [comment]
> Also I noticed that if you move them item (drag), the time will disappear.

I think that happens with the forge tooltip in general

**Comment by FrutyX** [comment]
You right.

**Comment by FrutyX** [comment]
2 secs difference now :P.

**Comment by GreatApo** [comment]
> 2 secs difference now :P.

Yeah, that's not going to be exact :P

**Comment by FrutyX** [comment]
This has to be related, it's new, it happened in packages, amulets only section.
![error](https://user-images.githubusercontent.com/23278552/176386164-f850e208-3f35-474f-a32b-6df1b30746d3.PNG)



Could be something random, maybe when I was getting them, hard to say, can't trigger it, yet.

**Comment by SleepingShadow-bit** [comment]
Is it possible to implement the melting time in the advanced filter (#343)as well?  For example Less than 3h melting time

**Comment by FrutyX** [comment]
Yea, I guess, however the advanced filter is so outdated now, that no one dares to touch it :D.

**Comment by SleepingShadow-bit** [comment]
I also have to admit that I don't use the advanced filter.  I never need the search option that is implemented :) However, there are very good ideas in the topic #343

**Comment by GreatApo** [comment]
> This has to be related, it's new, it happened in packages, amulets only section.
> 
> Could be something random, maybe when I was getting them, hard to say, can't trigger it, yet.

Not related, I improved the code to refresh the tooltip when moving items. It must be fixed now.

> Is it possible to implement the melting time in the advanced filter as well?

I will post it under the relevant issue

**Comment by FrutyX** [comment]
Implemented in v4.3.6, closing, re-open if needed.
